lib/src/feature/authentication/data/repositories/authentication_repository_impl.dart

```diff
@@ -1,29 +1,41 @@
+import 'dart:developer';
+
+import 'package:flutter_template/src/core/service/network/rest_client.dart';
+
 import '../../domain/entities/login_entity.dart';
 import '../../domain/entities/sign_up_entity.dart';
 import '../../domain/repositories/authentication_repository.dart';
-import '../data_sources/authentication_data_source.dart';
 import '../models/login_model.dart';
-import '../models/sign_up_model.dart';
 
 final class AuthenticationRepositoryImpl extends AuthenticationRepository {
-  final AuthenticationDataSource remote;
+  AuthenticationRepositoryImpl({
+    required this.restClient,
+  });
 
-  AuthenticationRepositoryImpl({required this.remote});
+  final RestClient restClient;
 
   @override
   Future<SignUpResponseEntity> register(SignUpRequestEntity data) async {
-    return await request(() async {
-      final response = await remote.register(data.toJson());
-      return SignUpResponseModel.fromJson(response.data);
-    });
+    // TODO: implement resetPassword
+    throw UnimplementedError();
   }
 
   @override
   Future<LoginResponseEntity> login(LoginRequestEntity data) async {
-    return await request(() async {
-      final response = await remote.login(data.toJson());
-      return LoginResponseModel.fromJson(response.data);
-    });
+    try {
+      final response = await restClient.login(
+        LoginRequest(
+          username: data.username,
+          password: data.password,
+        ),
```

Comment:
If we properly implement the Entity Model relationship, then we can pass the data without modification, I think.